### PR TITLE
Add feature/canvas

### PR DESCRIPTION
**Pull request by priyankj-simform**
Add - Canvas examples of compose.

### CODE REVIEW

File: app/src/main/java/com/jetpack/compose/learning/canvas/BasicCanvasExampleActivity.kt [hunk 1]
```diff
@@ -0,0 +1,345 @@
...
+                        navigationIcon = {
+                            IconButton(onClick = { onBackPressed() }) {
+                                Icon(Icons.Filled.ArrowBack, contentDescription = null)
+                            }
+                        }
+                    )
+                }) {
+
+                    val scrollState = rememberScrollState()
+
```
Comment (yashwantgowla-simform) [inline app/src/main/java/com/jetpack/compose/learning/canvas/BasicCanvasExampleActivity.kt:70]: Please remove empty line

File: app/src/main/java/com/jetpack/compose/learning/canvas/BasicCanvasExampleActivity.kt [hunk 2]
```diff
@@ -0,0 +1,345 @@
...
+                }
+            }
+        }
+    }
+
+    @Composable
+    fun BasicCanvasExample(modifier: Modifier = Modifier) {
+
+        val radiusAnimation = remember { Animatable(0f) }
+
```
Comment (yashwantgowla-simform) [inline app/src/main/java/com/jetpack/compose/learning/canvas/BasicCanvasExampleActivity.kt:97]: Please remove empty line

File: app/src/main/java/com/jetpack/compose/learning/canvas/BasicCanvasExampleActivity.kt [hunk 3]
```diff
@@ -0,0 +1,345 @@
...
+                )
+            }
+
+            inset(50f) {
+                drawArc(Color.Blue, 0f, radiusAnimation.value, false, style = stroke)
+            }
+        }
+    }
+
+    @Composable
```
Comment (yashwantgowla-simform) [inline app/src/main/java/com/jetpack/compose/learning/canvas/BasicCanvasExampleActivity.kt:148]: Please add method documentation.

File: app/src/main/java/com/jetpack/compose/learning/canvas/CanvasTextImageActivity.kt [hunk 4]
```diff
@@ -0,0 +1,355 @@
...
+                    dstSize = IntSize(dstWidth, dstHeight),
+                )
+            }
+        }
+
+        Text(text = "srcOffsetX $srcOffsetX")
+        Slider(
+            value = srcOffsetX.toFloat(),
+            onValueChange = { srcOffsetX = it.toInt() },
+            valueRange = -300f..300f,
```
Comment (yashwantgowla-simform) [inline app/src/main/java/com/jetpack/compose/learning/canvas/CanvasTextImageActivity.kt:230]: Please add values in constants

File: app/src/main/java/com/jetpack/compose/learning/canvas/CanvasTouchOperationActivity.kt [hunk 5]
```diff
@@ -0,0 +1,680 @@
...
+                        }
+                    }
+                }
+            }
+
+        Canvas(modifier = modifier) {
+            when (drawEvent) {
+                DrawEvent.DOWN -> {
+                    currentPath.moveTo(currentPosition.x, currentPosition.y)
+
```
Comment (yashwantgowla-simform) [inline app/src/main/java/com/jetpack/compose/learning/canvas/CanvasTouchOperationActivity.kt:174]: Please remove empty line

File: app/src/main/java/com/jetpack/compose/learning/canvas/CanvasTouchOperationActivity.kt [hunk 6]
```diff
@@ -0,0 +1,680 @@
...
+                }
+                DrawEvent.MOVE -> {
+                    currentPath.quadraticBezierTo(
+                        previousPosition.x,
+                        previousPosition.y,
+                        (previousPosition.x + currentPosition.x) / 2,
+                        (previousPosition.y + currentPosition.y) / 2
+
+                    )
+
```
Comment (yashwantgowla-simform) [inline app/src/main/java/com/jetpack/compose/learning/canvas/CanvasTouchOperationActivity.kt:185]: Please remove empty line

**Review by yashwantgowla-simform — DISMISSED** [review]

File: app/src/main/java/com/jetpack/compose/learning/canvas/CanvasAndroidEasterEggActivity.kt [hunk 7]
```diff
@@ -0,0 +1,200 @@
...
+        }
+    }
+
+    private fun checkAngle(newAngle: Float, innerCircleAngle: Float): Boolean {
+        return abs(newAngle - innerCircleAngle) < maxAngleChange
+    }
+
+    private fun adjustAngle(angle: Float): Float {
+        var actualAngle = 360 - angle
+        if (actualAngle < startAngle)
```
Comment (nikunj-b-simform) [inline app/src/main/java/com/jetpack/compose/learning/canvas/CanvasAndroidEasterEggActivity.kt:184]: Conditional statements are always required to be enclosed with braces.

File: app/src/main/java/com/jetpack/compose/learning/canvas/CanvasAndroidEasterEggActivity.kt [hunk 8]
```diff
@@ -0,0 +1,200 @@
...
+        }
+        val measureText = paint.measureText(mainText)
+        val modifier = Modifier
+            .fillMaxWidth()
+            .fillMaxHeight()
+            .background(Color(0xFF073042))
+            .pointerInteropFilter {
+                when (it.action) {
+                    MotionEvent.ACTION_DOWN -> {
+
```
Comment (nikunj-b-simform) [inline app/src/main/java/com/jetpack/compose/learning/canvas/CanvasAndroidEasterEggActivity.kt:118]: Seems like an unnecessary action.

**Review by nikunj-b-simform — DISMISSED** [review]

**Review by yashwantgowla-simform — APPROVED** [review]

**Pull request merged**